intake/source/utils.py

```diff
@@ -210,6 +212,9 @@ def path_to_glob(path):
 
     fmt = Formatter()
 
+    if not isinstance(path, str):
+        return path
```

Comment:
Do you think it's worth being more careful here, checking for specific types (tuple, list), that it's iterable, or something else?

Reply:
I don't just because this function is called right away on the urlpath and is only useful for converting from something like `data{field}.csv` to `data*.csv`. In any plugin using this path as a pattern approach, we would check if `path_to_glob(urlpath) == urlpath` to decide if there were any pattern fields in the path. So by just returning path we ensure that the path won't be treated as a pattern.

Reply:
OK, I can agree with that.